docs/TransparentLogging.md

```diff
@@ -0,0 +1,280 @@
+
+For our examples:
+ - A Certificate Transparency log requires entries to have X.509 structure, and
+   trusted-roots authentication.
+ - A Binary Transparency log requires entries to have the structure of an
+   installable binary package, and vendor-signing key authentication.
+ - A Pastebin log has little structural requirement and no authentication
+   requirement, and so is vulnerable to different attacks (swamping, uploading
+   illegal content etc.)
+
+Finally, note that a transparent Log normally acts as an **observatory, not as a
+police officer**, in the overall ecosystem.  With this in mind, it's often
+sensible for the structure checks on submissions to be lax, so that
+technically-invalid objects that are still signed and distributed can be
+monitored and attributed.
+
+ - A Certificate Transparency Log may well accept X.509 certificates that fail
+   various sorts of validity check, as long as the signature validation is
+   correct.  This means that certificates that might be accepted by lax clients
+   are still logged, and certificate authorities that issue invalid certificates
+   can be held to account more easily.
+
+## Inclusion Proofs vs. Promises
+
+Transparent Logs have two distinct mechanisms for guaranteeing inclusion of a
+particular entry in the Log.
+
+ - An **inclusion proof** gives cryptographic proof that a particular Merkle
+   tree head includes a given entry: the entry's hash and the proof hashes can be
+   combined to recreate the root hash of the tree.
+ - An **inclusion promise** is a signed commitment from the Log to incorporate a
+   particular entry within a defined time period.  This requires a separate
+   subsequent check to confirm that the Log does indeed fulfil this promise.
+
+A related factor is the number of entries that are incorporated into each new
+tree head issued by the Log, which we'll call the tree head **batch size**.
+
+ - Logs that issue inclusion promises can batch together multiple submissions
+   (e.g. an hour's worth) into a new tree head.  This is more likely to produce
+   large batch sizes, but is not guaranteed to (e.g. if submission rates are
+   very low).
+ - Logs that immediately issue inclusion proofs are likely to have smaller batch
+   sizes, either because they issue a new tree head per submission (batch size =
+   1), or because the amount of batching possible while still responding to
+   submissions quickly is small.
+
+The batch size per new tree head is important for privacy reasons: a small batch
+size, particularly a batch of size 1, means that a new signed tree head
+correlates directly with a single submission.  This in turn means that a user
+requesting proofs to/from that tree head, or gossiping that tree head, is likely
+to be interested in that specific entry in the Log.
+
+So, if privacy is a concern then:
+ - if submission rates are very low, then using inclusion promises with a large
+   batch size is most likely to preserve privacy
```

Comment:
Again this depends on the rest of the requirements - if sufficient latency in integration is acceptable then the retry/proof option is fine here too, and may simplify other parts of the system since they only have the one type of proof to deal with.

You touch on these points below, but just calling it out here to make a small edit if you do so above too.